marshmallow_jsonschema/base.py

```diff
@@ -254,7 +254,7 @@ def _get_python_type(self, field):
     def _get_schema_for_field(self, obj, field):
         """Get schema and validators for field."""
         if hasattr(field, "_jsonschema_type_mapping"):
-            schema = field._jsonschema_type_mapping()
+            schema = field._jsonschema_type_mapping(self, obj)
```

Comment:
Hmm, the problem with this is that it would break backward compatibility with custom type support as currently documented:

https://github.com/fuhrysteve/marshmallow-jsonschema#custom-type-support

Is there a way we can achieve what your trying to do without breaking backward compatibility?

Reply:
hmm yeah, i think a part of the difficulty here is actually the kind of ad-hoc nature of `jsonschema_type_mapping`, where there's no explicit definition of an interface so we can't just update some base definition like 

```
def _jsonschema_type_mapping(self, json_schema = None, obj = None):
  ...
```
to make this easily work for everyone.

I think similar to how you're using reflection to check if that `_jsonschema_type_mapping` even exists, I can use `inspect` to count the number of arguments and only pass the context in if the argument count supports it. This would preserve existing behavior but does make this interface a bit more loosey than it already is.

I think I'm ok with that. You could argue a proper solution requires a revamp of how this functionality is supported anyhow, so the best we can do atm is just to bolt more on to what was bolted on.

I'll make an update, and also be sure to update the README this time.